Gems/Atom/Feature/Common/Code/Source/ImGui/ImGuiPass.cpp

```diff
@@ -67,8 +67,8 @@ namespace AZ
 
         ImGuiPass::ImGuiPass(const RPI::PassDescriptor& descriptor)
             : Base(descriptor)
-            , AzFramework::InputChannelEventListener(AzFramework::InputChannelEventListener::GetPriorityUI())
-            , AzFramework::InputTextEventListener(AzFramework::InputTextEventListener::GetPriorityUI())
+            , AzFramework::InputChannelEventListener(AzFramework::InputChannelEventListener::GetPriorityUI() + 1) // Prioritize ImGui input over in-game UI such as LyShine
```

Comment:
Should we consider adding a `GetPriorityImGUI()` function that encapsulates this (in case priorities shift in the future)?

Reply:
> Should we consider adding a `GetPriorityDebugUI()` function that encapsulates this (in case priorities shift in the future)?

That's tricky because I'm not sure if ImGui is generic enough to warrant being predefined in AZ.

Reply:
Yep I originally typo'd GetPriorityDebugUI as GetPriorityImGUI and ninja-edited it immediately after, you didn't misread.

Reply:
These are the current pre-defined values:

        ////////////////////////////////////////////////////////////////////////////////////////////
        ///@{
        //! Predefined input event listener priority, used to sort handlers from highest to lowest
        inline static AZ::s32 GetPriorityFirst()    { return std::numeric_limits<AZ::s32>::max(); }
        inline static AZ::s32 GetPriorityDebug()    { return (GetPriorityFirst() / 4) * 3;        }
        inline static AZ::s32 GetPriorityUI()       { return GetPriorityFirst() / 2;              }
        inline static AZ::s32 GetPriorityDefault()  { return 0;                                   }
        inline static AZ::s32 GetPriorityLast()     { return std::numeric_limits<AZ::s32>::min(); }
        ///@}
        
which are pretty arbitrary (basically just what seemed sensible to me at the time).

Guess it just comes down to whether we would ever expect ImGuiPass to be used in a non-debug context?

Was talking with michabr about this specific issue earlier, and seems like ImGuiPass falls somewhere between GetPriorityDebug and GetPriorityUI, so creating a new GetPriorityDebugUI makes sense to me and might be a good solution here. Using GetPriorityDebug should also work fine, which is what the ImGuiManager currently uses (although I'd go with -1 instead of +1 in this case).